### PR TITLE
207 update install instructions

### LINKED ISSUE #207: User install instruction out of sync with code
Top level README.md currently reads

```
[![Python 2.7](https://img.shields.io/badge/python-2.7-blue.svg)](https://www.python.org/downloads/release/python-2715/) is (currently) the Python version needed for running/installing FitBenchmarking. Instructions for setting up python are available [here](https://github.com/mantidproject/fitbenchmarking/wiki/Setting-up-Python).

For help of how to use the command line/terminal, click the hyperlink corresponding to your operating system: [Windows](https://red-dot-geek.com/basic-windows-command-prompt-commands/), [macOS](http://newsourcemedia.com/blog/basic-terminal-commands/) and [Ubuntu](https://hackingpress.com/basic-linux-commands/)

1. Download this repository or clone it using [git](https://git-scm.com/):
`git clone https://github.com/mantidproject/fitbenchmarking.git`
2. Open up a terminal (command prompt) and go into the `fitbenchmarking` directory.
3. Once you are in the right directory, type `python setup.py install`.
4. Install mantid, platform specific instructions [here](https://github.com/mantidproject/fitbenchmarking/wiki/Installing-Mantid).
5. Open the mantidpython terminal
    * Ubuntu: in a normal terminal cd to /opt/Mantid/bin and run `./mantidpython`
    * Windows: go to where you installed Mantid and search for a file called `mantidpython`
    * macOS: in a normal terminal run `/Applications/MantidPlot.app/Contents/MacOS/mantidpython`
```

i.e. it is assumed Mantid is required to be installed, but that is no longer the case.

Further with #199 sasview, and shipping sascalc + installing bumps and sasmodels with setup.py,  the part of sasview needed to test its minimizers are in fact by default, like for scipy installed by default.

* test that setup.py install works with bumps (sasview) minimizers

If yes update 

* README.md accordingly
* should example_runScripts_SasView.py be deleted and instead in example_runScripts.py change software = ['scipy'] to software = ['scipy', 'sasview']?

Finally a suggestion is to reorganise the Quick Start section of the README.md to focus on running with scipy (and sasview), and explain further down how to get it working with Mantid and any other future software that needs specifical install instructions

### LINKED ISSUE #268: IndexError while running multiple software packages
**Description of the error**
Observed an `IndexError` whilst running `mantid` and `scipy` in master:
Error

**Describe the expected result**
Re-enable comparison of multiple software packages

**Suggested fix**
In the merge of #225, ` fitbenchmarking/fitting_benchmarking.py` was change so I suspect the issue lies here.

### PR DESCRIPTION
**Pull request by AndrewLister-STFC**
#### Description of Work

Fixes #207 
Fixes #268

Updates the root level README.md, and the setup.py to allow a completely empty python environment.

Testing required:

* New wording OK
* Check install works on mac, windows and ubuntu

### CODE REVIEW

File: example_scripts/example_runScripts.py [hunk 1]
```diff
@@ -21,20 +22,57 @@
 fitbenchmarking_folder = os.path.abspath(os.path.join(current_path, os.pardir))
 scripts_folder = os.path.join(fitbenchmarking_folder, 'fitbenchmarking')
 sys.path.insert(0, scripts_folder)
+sys.path.insert(1, fitbenchmarking_folder)
```
Comment (Anders-Markvardsen) [inline example_scripts/example_runScripts.py]: Probably not necessary

**Comment by Anders-Markvardsen** [comment]
I (@Anders-Markvardsen) will test install works on Windows

**Comment by Anders-Markvardsen** [comment]
Install works on Windows

**Comment by arm61** [comment]
please ignore my stupidity in messing with this

**Comment by AndrewLister-STFC** [comment]
Tested on:

- [ ] Windows

- [ ] Ubuntu

- [ ] Mac

**Comment by AndrewLister-STFC** [comment]
From @tyronerees, we should change the following: 

`python setup.py install` -> `python setup.py install --user`

remove package_data (needs to check this works)

File: README.md [hunk 2]
```diff
@@ -11,66 +11,56 @@ FitBenchmarking is a tool for comparing different minimizers/fitting software ba
...
 
-## FitBenchmarking Scipy
-The `example_runScripts.py` file is designed to benchmark minimizers supported by software/libraries that provide straightforward cross-platform Python install; as of know this mean SciPy (more details provided in the file itself).
-
-For this to work scipy version 0.17 or higher is needed (which includes needed [curve_fit](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html) support). **The Linux distributions we have tested against so far have all included scipy 0.17+ (0.17 is from Feb 2016).**
+1. Download this repository or clone it using [git](https://git-scm.com/): \
+    `git clone https://github.com/fitbenchmarking/fitbenchmarking.git`
+2. Open up a terminal (command prompt) and go into the `fitbenchmarking` directory.
+3. Once you are in the right directory, start the install with:\
+    `python setup.py install`
```
Comment (tyronerees) [inline README.md]: I need to append `--user` to get this to work without needing sudo on Ubuntu

File: setup.py [hunk 3]
```diff
@@ -11,8 +11,10 @@
       url='http://github.com/fitbenchmarking/fitbenchmarking',
       license='GPL-3.0',
       packages=find_packages(),
-      install_requires=['docutils', 'scipy>=0.17,<1.3', 'bumps', 'sasmodels', 'lxml'],
+      install_requires=['docutils', 'numpy<1.17', 'matplotlib<3.0',
+                        'scipy>=0.17,<1.3', 'bumps', 'sasmodels', 'lxml', 'h5py'],
```
Comment (tyronerees) [inline setup.py]: `scipy=0.17` fails for me on Ubuntu (issue #128) -- I'd like to see this be `scipy>=0.18`

File: README.md (hunk 2 above)
Comment (AndrewLister-STFC) [reply README.md]: Fixed

File: example_scripts/example_runScripts.py (hunk 1 above)
Comment (AndrewLister-STFC) [reply example_scripts/example_runScripts.py]: The sys.path calls are being removed in #240

File: setup.py (hunk 3 above)
Comment (AndrewLister-STFC) [reply setup.py]: Done!

Comment (tyronerees) [reply setup.py]: Cool -- this now fixes #128 (adding this in the hope GitHub will close that when this is merged)

@Anders-Markvardsen @AnthonyLim23 : please shout if there is a hidden reason why we should be supporting scipy 0.17

**Comment by AndrewLister-STFC** [comment]
Debugging this looks like its a hdf5/mantid communication issue.

Mantid is compiled with hdf5 ver 1.8.16, whereas the version pip installs is 1.10.4. So it fails with a warning.

```
/usr/local/lib/python2.7/dist-packages/h5py-2.10.0-py2.7-linux-x86_64.egg/h5py/__init__.py:40: UserWarning: h5py is running against HDF5 1.8.16 when it was built against 1.10.4, this may cause problems
  '{0}.{1}.{2}'.format(*version.hdf5_built_version_tuple)
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.10.4, library is 1.8.16
```

**Comment by TOFarmer** [comment]
Installation instructions work on Mac.  

Only comment would be that changing to `python setup.py install --user` is fine as long as you are not using virtualenv, in which case you must exclude `--user`.  Not sure if you want to include a note about this in installation instructions.

Does this still need testing on Windows? @Anders-Markvardsen

**Comment by tyronerees** [comment]
This will resolve #268

**Comment by tyronerees** [comment]
Closing and re-opening to try to trigger coveralls to finish

**Comment by Anders-Markvardsen** [comment]
reminder to @Anders-Markvardsen to test on windows

**Pull request merged**